pywinauto/win32structures.py

```diff
@@ -51,41 +51,41 @@ def __str__(self):
 
         fields in exceptList will not be printed"""
         lines = []
-        for f in self._fields_:
+        for f in getattr(self, "_fields_", list()):
             name = f[0]
             lines.append("%20s\t%s"% (name, getattr(self, name)))
 
         return "\n".join(lines)
 
     #----------------------------------------------------------------
-    def __eq__(self, other_struct):
+    def __eq__(self, other):
         """Return True if the two structures have the same coordinates"""
-        if isinstance(other_struct, Struct):
+        fields = getattr(self, "_fields_", list())
+        if isinstance(other, Struct):
             try:
                 # pretend they are two structures - check that they both
                 # have the same value for all fields
-                are_equal = True
-                for field in self._fields_:
+                if len(fields) != len(getattr(other, "_fields_", list())):

```

Comment:
The second operand could be just `len(fields)` so that `getattr` is not called twice.

Reply:
I'm afraid I don't understand this one.